FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -696,11 +759,189 @@ static MQTTStatus_t prvMQTTConnect( MQTTContext_t * pxMQTTContext,
+    RetryUtilsParams_t xReconnectParams;
+
+    #if defined( democonfigUSE_TLS ) && ( democonfigUSE_TLS == 1 )
+        TlsTransportStatus_t xNetworkStatus;
+        NetworkCredentials_t xNetworkCredentials = { 0 };
+
+        /* Set the credentials for establishing a TLS connection. */
+        xNetworkCredentials.pRootCa = ( const unsigned char * ) democonfigROOT_CA_PEM;
+        xNetworkCredentials.rootCaSize = sizeof( democonfigROOT_CA_PEM );
+        xNetworkCredentials.pClientCert = ( const unsigned char * ) democonfigCLIENT_CERTIFICATE_PEM;
+        xNetworkCredentials.clientCertSize = sizeof( democonfigCLIENT_CERTIFICATE_PEM );
+        xNetworkCredentials.pPrivateKey = ( const unsigned char * ) democonfigCLIENT_PRIVATE_KEY_PEM;
+        xNetworkCredentials.privateKeySize = sizeof( democonfigCLIENT_PRIVATE_KEY_PEM );
+    #else /* if defined( democonfigUSE_TLS ) && ( democonfigUSE_TLS == 1 ) */
+        PlaintextTransportStatus_t xNetworkStatus;
+    #endif /* if defined( democonfigUSE_TLS ) && ( democonfigUSE_TLS == 1 ) */
+
+    /* Initialize reconnect attempts and interval. */
+    xReconnectParams.maxRetryAttempts = MAX_RETRY_ATTEMPTS;
+    RetryUtils_ParamsReset( &xReconnectParams );
+
+    /* Attempt to connect to MQTT broker. If connection fails, retry after a
+     * timeout. Timeout value will exponentially increase until them maximum
+     * number of attempts are reached.
+     */
+    do
+    {
+        /* Establish a TCP connection with the MQTT broker. This example connects to
+         * the MQTT broker as specified in democonfigMQTT_BROKER_ENDPOINT and
+         * democonfigMQTT_BROKER_PORT at the top of this file. */
+        LogInfo( ( "Create a TCP connection to %s:%d.",
+                   democonfigMQTT_BROKER_ENDPOINT,
+                   democonfigMQTT_BROKER_PORT ) );
+
+        #if defined( democonfigUSE_TLS ) && ( democonfigUSE_TLS == 1 )
+            xNetworkStatus = TLS_FreeRTOS_Connect( pxNetworkContext,
+                                                   democonfigMQTT_BROKER_ENDPOINT,
+                                                   democonfigMQTT_BROKER_PORT,
+                                                   &xNetworkCredentials,
+                                                   mqttexampleTRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                                   mqttexampleTRANSPORT_SEND_RECV_TIMEOUT_MS );
+            xConnected = ( xNetworkStatus == TLS_TRANSPORT_SUCCESS );
+        #else
+            xNetworkStatus = Plaintext_FreeRTOS_Connect( pxNetworkContext,
+                                                         democonfigMQTT_BROKER_ENDPOINT,
+                                                         democonfigMQTT_BROKER_PORT,
+                                                         mqttexampleTRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                                         mqttexampleTRANSPORT_SEND_RECV_TIMEOUT_MS );
```

Comment:
Maybe define the transport timeout depending on whether a plaintext vs TLS connection is used.

Reply:
Does it need to be different? We use 200 for both the plaintext and mutual auth demos. It was 20 before was that I was only testing against localhost, and I forgot to increase it